cms/core/tests/test_video_embed_block.py

```diff
@@ -0,0 +1,254 @@
+from django.test import TestCase
+from wagtail.blocks import StructBlockValidationError
+from wagtail_factories import ImageFactory
+
+from cms.core.blocks import (
+    VideoEmbedBlock,
+)
+
+
+class VideoEmbedBlockTestCase(TestCase):
+    """Test for video embed block."""
+
+    def test_videoembedblock_clean__invalid_domain(self):
+        """Check the VideoEmbedBlock validates the supplied URL."""
+        block = VideoEmbedBlock()
+        image = ImageFactory.create()
```

Comment:
Move this to the `setUpTestData` class method so it is not created for every single sub-test

```
    @classmethod
    def setUpTestData(cls):
        cls.image = ImageFactory()
```

and it should help avoid repetion too

Reply:
Have done this now

Reply:
moving them to `setUp` avoids the repetition in the individual tests, but moving them to `setUpTestData` classmethod:

```python
@classmethod
def setUpTestData(cls)
   cls.block = VideoEmbedBlock()
   cls.image = ImageFactory() 
```

https://adamj.eu/tech/2021/04/12/how-to-convert-a-testcase-from-setup-to-setuptestdata/

But this is not a hill I will die on :)

Reply:
Ah thank you - TIL. Now updated.